core/commands/swarm.go

```diff
@@ -476,6 +507,21 @@ func peersWithAddresses(addrs []string) (pis []pstore.PeerInfo, err error) {
 	return pis, nil
 }
 
+func getNode(req cmds.Request, res cmds.Response) *core.IpfsNode {
```

Comment:
When I see copy-paste, I have the urge to extract and name the thing. Since all commands in this file wanted an online node and handled the error the same way, this seemed to be the logical step to reduce coupling. I even played with the idea to look further into other commands and introduce a middle-man on the req itself, but I stopped myself.

I can get rid of the res parameter, but then I would return `(*core.IpfsNode, commands.Error)` and add a `func (*commands.response) SetError(*Error)` if that is fine with you.